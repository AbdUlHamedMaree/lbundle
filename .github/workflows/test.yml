name: Test

on:
  pull_request:
    branches:
      - main
      - staging
      - dev
      - release/*

jobs:
  test:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v4

      - name: Setup Node.js
        uses: actions/setup-node@v4
        with:
          node-version-file: .nvmrc

      - name: Setup Bun
        uses: oven-sh/setup-bun@v1
        with:
          bun-version: latest

      - name: Install dependencies
        run: bun i --frozen-lockfile

      - name: Lint
        run: bun run lint

      - name: Test
        run: bun run test
