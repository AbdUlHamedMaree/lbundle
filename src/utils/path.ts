import path from 'path';

export function normalizePath(...paths: string[]): string {
  const f = path.join(...paths).replaceAll('\\', '/');

  if (/^\.[/\\]/.test(paths[0])) return `./${f}`;

  return f;
}
