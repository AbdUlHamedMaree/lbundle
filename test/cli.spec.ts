// eslint-disable-next-line import/no-unresolved
import { expect, it, describe } from 'bun:test';
import path from 'path';
import fs from 'fs';
import targetPkg from './project/package.json';

const __dirname = import.meta.dir;

const rootDir = path.resolve(__dirname, '..');

const cleanDist = () =>
  fs.promises.rm(path.resolve(__dirname, 'project', 'dist'), {
    force: true,
    recursive: true,
  });

const cleanAndBuild = async (args: string[] = []) => {
  await cleanDist();

  const proc = Bun.spawn(
    ['bun', '--bun', './src/cli.ts', '--cwd', './test/project', ...args],
    {
      cwd: rootDir,
      stdout: 'inherit',
      stderr: 'inherit',
    }
  );

  const code = await proc.exited;

  console.log('pizza', code)

  if (code > 0) {
    console.error('[lbundle] Failed to build');
    process.exit(code);
  }
};

const getBuildFileContent = (fileName: string) => {
  const file = Bun.file(path.resolve(__dirname, 'project', 'dist', fileName));

  return file.text();
};

describe('cli option', () => {
  it('no options', async () => {
    await cleanAndBuild();

    const indexJs = await getBuildFileContent('index.js');

    expect(indexJs).toMatchSnapshot('default build');
  });
});
