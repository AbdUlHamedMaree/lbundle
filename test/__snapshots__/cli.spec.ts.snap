// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`React Library Build with no options: index.js 1`] = `
"'use strict';

var index = require('./index2.js');



module.exports = index;
//# sourceMappingURL=index.js.map
"
`;

exports[`React Library Build with no options: index.mjs 1`] = `
"export { Button } from './components/button/index.mjs';
//# sourceMappingURL=index.mjs.map
"
`;

exports[`React Library Build with no options: index.d.ts 1`] = `
"import './index.scss';
export * from './components';
"
`;

exports[`React Library Build with no options: components/index.d.ts 1`] = `
"export * from './button';
"
`;

exports[`React Library Build with no options: components/button/index.js 1`] = `
"'use strict';

var index = require('../../_virtual/index3.js');
var require$$0 = require('react/jsx-runtime');
var styles_module = require('../../_virtual/styles.module.scss.js');

(function (exports) {
	Object.defineProperty(exports, "__esModule", {
	    value: true
	});
	Object.defineProperty(exports, "Button", {
	    enumerable: true,
	    get: function() {
	        return Button;
	    }
	});
	const _jsxruntime = require$$0;
	const _stylesmodulescss = /*#__PURE__*/ _interop_require_default(styles_module);
	function _interop_require_default(obj) {
	    return obj && obj.__esModule ? obj : {
	        default: obj
	    };
	}
	const Button = (param)=>{
	    let { children } = param;
	    return (0, _jsxruntime.jsx)("button", {
	        className: _stylesmodulescss.default.button,
	        children: children
	    });
	}; 
} (index.__exports));
//# sourceMappingURL=index.js.map
"
`;

exports[`React Library Build with no options: components/button/index.mjs 1`] = `
"import { jsx } from 'react/jsx-runtime';
import modules_efc4e723 from './styles.module.scss.mjs';

const Button = (param)=>{
    let { children } = param;
    return jsx("button", {
        className: modules_efc4e723.button,
        children: children
    });
};

export { Button };
//# sourceMappingURL=index.mjs.map
"
`;

exports[`React Library Build with no options: components/button/index.d.ts 1`] = `
"export type ButtonProps = {
    children?: React.ReactNode;
};
export declare const Button: React.FC<ButtonProps>;
"
`;

exports[`Utils Library Build with no options: cli.js 1`] = `
"#!/usr/bin/env node
'use strict';

var require$$0 = require('lodash.merge');

var cli = {};

var src = {};

(function (exports) {
	Object.defineProperty(exports, "__esModule", {
	    value: true
	});
	Object.defineProperty(exports, "myMerge", {
	    enumerable: true,
	    get: function() {
	        return myMerge;
	    }
	});
	const _lodashmerge = /*#__PURE__*/ _interop_require_default(require$$0);
	function _interop_require_default(obj) {
	    return obj && obj.__esModule ? obj : {
	        default: obj
	    };
	}
	const myMerge = (o1, o2)=>(0, _lodashmerge.default)(o1, o2); 
} (src));

Object.defineProperty(cli, "__esModule", {
    value: true
});
const _ = src;
const o = (0, _.myMerge)({
    a: 1,
    b: 1
}, {
    b: 2,
    c: 2
});
console.log(JSON.stringify(o, null, 2));

module.exports = cli;
//# sourceMappingURL=cli.js.map
"
`;

exports[`Utils Library Build with no options: index.js 1`] = `
"'use strict';

var index = require('./index2.js');



module.exports = index;
//# sourceMappingURL=index.js.map
"
`;

exports[`Utils Library Build with no options: index.mjs 1`] = `
"import merge from 'lodash.merge';

const myMerge = (o1, o2)=>merge(o1, o2);

export { myMerge };
//# sourceMappingURL=index.mjs.map
"
`;

exports[`Utils Library Build with no options: index.d.ts 1`] = `
"export declare const myMerge: (o1: object, o2: object) => object;
"
`;
